robocop/checkers/spacing.py

```diff
@@ -75,24 +75,12 @@
         severity=RuleSeverity.WARNING,
     ),
     "1008": Rule(
-        RuleParam(
```

Comment:
If we're going to remove parameters we will need to make next release major release (because people configuration will break). We could also introduce deprecation period but it would require extra work, may not be worth it

Reply:
After some debugging, for example if I run tests on all_passing.robot file in suite with all tests, bad-indent will be configured to 3 spaces. Somehow pytest remebers the last rule configuration and uses it in different test (even if we're using totally new Robocop instance)

Reply:
Updating last test in bad-indent fix 2 failing tests:

```
    @pytest.mark.parametrize("file_suffix, target_version", [("", ">=5"), ("_rf4", "==4.*"), ("_rf3", "==3.*")])
    def test_rule(self, file_suffix, target_version):
        self.check_rule(config="-c bad-indent:indent:4", expected_file=f"expected_output{file_suffix}.txt", target_version=target_version)

```
(we override previous -c with indent of 3 spaces)

Reply:
So we're talking about Robocop 3.0 here?

Reply:
Because I did not remove this rule, just scoped down its functionalities and moved to another rule. Previously, in such cases, we were bumping up the second number (Y) in X.Y.Z. versioning.

Reply:
Yes, Robocop 3.0. It is because if user have configuration that works for 2.8.0, it will result in exception (missing parameter) in 2.9.0 - we don't want that. Ie:

robocop -c bad-indent:strict=True . 

Will not work anymore

Reply:
Optionally you can leave the parameter, but don't use them and after deprecation period remove them

Reply:
Anyway, I think that even if I leave the parameter, we still should release 3.0.0, because of the changes to the columns for many rules. If someone used disablers, they may stop working, being misplaced now due to new locations, right?

Reply:
Yes

Reply:
Then we need to go with 3.0.0.

Should we add the parameters back for backward compatibility but just deprecate them and maybe remove in the next version? I think better not to break the day of the users 😉 As you said, we can add them and just don't call anywhere.

Reply:
I agree, some deprecation period would be graceful for users